chore: Rename dynamo.ingress to dynamo.frontend

grahamking opened the pull request:
We now start HTTP+pre-process+router like this:
```
python -m dynamo.frontend [--http-port 8080]
```

This fits the naming we agreed on. Previously the dynamo.frontend namespace was taken up by the examples, but those are gone now.


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->

## Summary by CodeRabbit

* **Documentation**
  * Updated frontend README to clarify node description and usage instructions.

* **Chores**
  * Updated internal references and build configuration to use the new frontend module path.

<!-- end of auto-generated comment: release notes by coderabbit.ai -->

rmccorm4 reviewed (approved):
LGTM after pre-commit fix

rmccorm4 commented on pyproject.toml:
```diff
@@ -79,7 +79,7 @@ requires = ["hatchling"]
 
 [tool.hatch.build.targets.wheel]
-packages = ["deploy/sdk/src/dynamo", "components/planner/src/dynamo", "components/ingress/src/dynamo", "components/backends/llama_cpp/src/dynamo"]
+packages = ["deploy/sdk/src/dynamo", "components/planner/src/dynamo", "components/frontend/src/dynamo", "components/backends/llama_cpp/src/dynamo"]
```
Given this change messing with the precommits notion of correct import order, I wonder if this is subtlely breaking the relative import of components.frontend - CC @indrajit96 @krishung5 in case this breaks anything, though I'm not sure of the current state of the multimodal example with the refactoring away from dynamo serve and such

rmccorm4 replied:
Yeah by no means a blocking comment, just an FYI that crossed my mind for the image and video enjoyers. Thanks for being on top of it 🚀

krishung5 commented on examples/multimodal/graphs/agg.py:
```diff
@@ -16,7 +16,8 @@
-from components.frontend import Frontend
 from components.processor import Processor
 
+from components.frontend import Frontend
```
Looks like for this change it doesn't change anything other than move the import to the other line. Is this required?

grahamking replied:
The `pre-commit` hook insisted. :shrug:

grahamking committed:
chore: Rename dynamo.ingress to dynamo.frontend

We now start HTTP+pre-process+router like this:
```
python -m dynamo.frontend [--http-port 8080]
```

This fits the naming we agreed on. Previously the dynamo.frontend namespace was taken up by the examples, but those are gone now.
